### ISSUE TITLE
User story - Profile page

### ISSUE THREAD

**Issue by stephendawsondev** [stephendawsondev/StackPortfolio#15]
### If applicable, add a related Epic.

Epic: #14 

### User Story

As a **user**, I can **have a profile page** so that I can **display my details and projects/jobs**.

### Acceptance Criteria

- The user should have a way to navigate to their profile page
- The profile page should have the user's profile image, name and other details they signed up with
- Recruiters should have their jobs displayed
- Tech users should have their projects displayed

### Please ensure you've completed these tasks before marking the issue as done.

- [x] Reviewed the user story for clarity and completeness.
- [x] Identified and linked related issues or epics.
- [x] Created or updated the necessary documentation.
- [x] Written unit and/or integration tests for functionality.
- [x] All code changes have been manually tested to ensure they meet the acceptance criteria.
- [x] Updated the issue with any relevant notes or feedback (to date).
- [x] Merged into the main branch and deployed to production (if applicable).

### Please add any further tasks that need to be completed before the issue can be marked as done.

- [x] Create the url in the backend for `/user/<slug>`
- [x] Create the template and UI
